Engines/Wine/QuickScript/Quick Script/script.js

```diff
@@ -135,6 +135,20 @@ module.default = class QuickScript {
         return this;
     }
 
+    /**
+     * determines which Wine version should be used
+     * required in case the version is computed by a function
+     * @param {wizard} wizard setup wizard (e.g. to show download progress of versions json)
+     * @returns {void}
+     */
+    _determineWineVersion(wizard) {
```

Comment:
I don't like that this method modifies `this._wineVersion`. I think it should return the extracted wine version instead of overriding `this._wineVersion`.

Reply:
Then you can never use `this._wineVersion` anywhere in the QuickScripts. I don't think that's very intuitive. I agree that the current handling is a bit strange but I don't like your solution either.

Reply:
Why?

My suggestion is to introduce a local variable `selectedWineVersion` or `resolvedWineVersion` that is used in the QuickScripts. This should not be that difficult because you are calling `_determineWineVersion` in the QuickScripts anyway right?

To better illustrate what I mean:

```javascript
const resolvedWineVersion = this._determineWineVersion(wizard);

// do something with resolvedWineVersion instead of this._wineVersion
```

Reply:
I can do that.